contracts/refs/CoreRefV2.sol

```diff
@@ -0,0 +1,245 @@
+        );
+        _;
+    }
+
+    // Modifiers to allow any combination of roles
+    modifier hasAnyOfTwoRoles(bytes32 role1, bytes32 role2) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfThreeRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfFourRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        bytes32 role4
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender) ||
+                _core.hasRole(role4, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfFiveRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        bytes32 role4,
+        bytes32 role5
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender) ||
+                _core.hasRole(role4, msg.sender) ||
+                _core.hasRole(role5, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    /// @notice set pausable methods to paused
+    function pause() public override onlyGuardianOrGovernor {
+        _pause();
+    }
+
+    /// @notice set pausable methods to unpaused
+    function unpause() public override onlyGuardianOrGovernor {
+        _unpause();
+    }
+
+    /// @notice address of the Core contract referenced
+    /// @return ICore implementation address
+    function core() public view override returns (ICoreV2) {
+        return _core;
+    }
+
+    /// @notice address of the Volt contract referenced by Core
+    /// @return IVolt implementation address
+    function volt() public view override returns (IVolt) {
+        return _core.volt();
+    }
+
+    /// @notice address of the Vcon contract referenced by Core
+    /// @return IERC20 implementation address
+    function vcon() public view override returns (IERC20) {
+        return _core.vcon();
+    }
+
+    /// @notice volt balance of contract
+    /// @return volt amount held
+    function voltBalance() public view override returns (uint256) {
+        return volt().balanceOf(address(this));
+    }
+
+    /// @notice vcon balance of contract
+    /// @return vcon amount held
+    function vconBalance() public view override returns (uint256) {
+        return vcon().balanceOf(address(this));
+    }
+
+    function _burnVoltHeld() internal {
+        volt().burn(voltBalance());
+    }
```

Comment:
I don't think we've used these anywhere, could remove to make the contracts smaller

Reply:
removed `_burnVoltHeld` as it is used nowhere. are you proposing we remove `voltBalance` and `vconBalance` also?

Reply:
`voltBalance` is used in the PSM, but we could refactor as there's no reason it has to have this logic. Bigger question is will we need either of these functions in the future?